libgadget/cooling.c

```diff
@@ -21,33 +21,55 @@
 #include <string.h>
 #include <math.h>
 #include <bigfile.h>
-
 #include "utils/endrun.h"
 #include "utils/mymalloc.h"
 #include "utils/interp.h"
 #include "physconst.h"
 #include "cooling.h"
 #include "cooling_rates.h"
+#include "cooling_qso_lightup.h"
 
 static struct cooling_units coolunits;
 
 /*Do initialisation for the cooling module*/
-void init_cooling(char * TreeCoolFile, char * MetalCoolFile, char * UVFluctuationFile, struct cooling_units cu, Cosmology * CP)
+void init_cooling(char * TreeCoolFile, char * MetalCoolFile, char * UVFluctuationFile, char * reion_hist_file, struct cooling_units cu, Cosmology * CP)
 {
     coolunits = cu;
 
     /*Initialize the cooling rates*/
     init_cooling_rates(TreeCoolFile, MetalCoolFile, CP);
     /*Initialize the uv fluctuation table*/
     init_uvf_table(UVFluctuationFile);
+    /* Initialize the helium reionization model*/
+    init_qso_lightup(reion_hist_file);
 }
 
 #define MAXITER 1000
 
+/* Wrapper function which returns the rate of change of internal energy in units of
+ * erg/s/g. Arguments:
+ * rho: density in protons/cm^3 (physical)
+ * u: internal energy in units of erg/g
+ * Z: metallicity
+ * redshift: redshift
+ * isionized: flags whether the particle has been HeII reionized.
+ */
+static double
+get_lambdanet(double rho, double u, double redshift, double Z, struct UVBG * uvbg, double * ne_guess, int isionized)
+{
+    double LambdaNet = get_heatingcooling_rate(rho, u, 1 - HYDROGEN_MASSFRAC, redshift, Z, uvbg, ne_guess);
```

Comment:
Does the meaning of this function change when you HeII to the picture? e.g. do we now have to be aware it is under an approximation, e.g. only the short mean free path heating?

Reply:
No, not at all: this is always the heating under the ionization equilibrium assumption, while the other code takes care of deviations from that.